package.json

```diff
@@ -14,8 +14,8 @@
     "bluebird": "3.4.1",
     "compressjs": "1.0.3",
     "gm": "1.23.1",
-    "handlebars": "^4.0.11",
-    "image-diff": "^2.0.0",
+    "handlebars": "4.0.11",
+    "image-diff": "2.0.0",
```

Comment:
Forgot to pin these to a specific version, otherwise package-lock.json doesn't lock them in fully. Kind of annoying.

Reply:
hm… what do you mean by "doesn't lock them in fully"? That'd be against the definition of package-lock:

> It describes the exact tree that was generated, such that subsequent installs are able to generate identical trees, regardless of intermediate dependency updates.

Reply:
You'd think that. For at least a few versions, NPM didn't follow its own definition: https://github.com/npm/npm/issues/18103

This *may* have been resolved since but I think plugging in specific version numbers is probably a good habit to get into.

Reply:
Oh my. I think that this comment - https://github.com/npm/npm/issues/18103#issuecomment-370401935 - captures my feelings about this revelation.

Reply:
I think Yarn does a better job of locking the versions?

https://yarnpkg.com/lang/en/docs/yarn-lock/

Reply:
@moollaza I'm not sure how much effort is involved, but it might be worth looking into later on. I get some joy out of the threads linking to that issue, though:

![screenshot from 2018-08-09 13-51-00](https://user-images.githubusercontent.com/1431577/43900091-b9136800-9bdb-11e8-9d7a-4177f3713cf4.png)

Reply:
@andrey-p you just need to install yarn and run `yarn`. It just works 👍 

https://yarnpkg.com/lang/en/docs/migrating-from-npm/

After that its `yarn add` instead of `npm install`

Reply:
yarn has it's own lock file (yarn.lock) and it's own set of problems + (by default) all requests go through fb servers. IMO we should research it properly before changing tools.

Reply:
> all requests go through fb servers.
Crap! Wasn't aware of that.